### PR TITLE
Implements Copilot() function

### PR DESCRIPTION
**Pull request by jas-valgotar**
This pull request introduces support for a new Copilot function in the Power Fx REPL, enabling integration with OpenAI's Responses API for text and structured output generation. It adds the necessary service interface, configuration extension, and REPL integration logic, along with comprehensive documentation and new tests for DType-to-JSON schema conversion.

**Copilot Integration:**

* Added the `ICopilotService` interface to define a contract for Copilot-like services, allowing asynchronous prompt/response interactions. [[1]](diffhunk://#diff-b6f8f6170355525bdb2c1810facb0738aa8bd0d9a0d563cbc4cf598bb2669decR1-R17) [[2]](diffhunk://#diff-8108e40f03cf00d4d24d73da9d1ad6bfb30c9fab9d9666bda9112e34b3073949R30)
* Introduced the `EnableCopilotFunction` extension method (marked as `[Obsolete("preview")]`) to `PowerFxConfig`, enabling the Copilot function in formula evaluation.
* Updated the REPL (`Program.cs`) to optionally load Copilot credentials, initialize the Copilot service, and inject it into the engine's service provider. The REPL now disposes of the Copilot service properly and provides user feedback if the service is unavailable. [[1]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7R31-R35) [[2]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7R127-R129) [[3]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7L171-R188) [[4]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7R202-R234) [[5]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7R261-R287) [[6]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7R311) [[7]](diffhunk://#diff-f7f431a9dc9547a2f1a690e640277bdf1cecbc5c42125ae6f7590769e768e0c7L283-R354)
* Added a detailed `COPILOT_README.md` with setup instructions, usage examples, troubleshooting, and extension guidance for the Copilot feature in the REPL.

**Testing and Validation:**

* Added a new test suite `DTypeToJsonSchemaTests` to verify the conversion of Power Fx types (`DType`) to JSON Schema, including edge cases and validation of output as valid JSON.

These changes lay the groundwork for Copilot-powered features in Power Fx, with clear extension points and robust testing for type schema generation.

### CODE REVIEW

**Comment by jas-valgotar** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Json/Functions/ICopilotService.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerFx.Core
+{
+    [ThreadSafeImmutable]
+    public interface ICopilotService : IDisposable
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Json/Functions/ICopilotService.cs:13]: Curious - why is this in Fx.Json and not Fx.Interpreter?

File: src/tools/Repl/COPILOT_README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+1. Copilot(prompt: Text): Text
+2. Copilot(prompt: Text, context: Any): Text
+3. Copilot(prompt: Text, context: Any, returnType: Type): ReturnType
+
+The structured overload requests a response that you can parse/interpret into the specified shape. (Actual enforcement still depends on your Copilot function implementation.)
+
+## Examples
+
+### Example 1: Simple Query
+```powerf
```
Comment (MikeStall) [inline src/tools/Repl/COPILOT_README.md]: nit - is this missing an 'x' at the end?

File: src/libraries/Microsoft.PowerFx.Json/Functions/ICopilotService.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Json/Functions/ICopilotService.cs:13]: Because we need to Convert the DType to JSON Schema, so LLM can understand that and respond in the JSON Schema. And convert the returned response in FormulaValue. All of that logic is in Fx.Json

File: src/libraries/Microsoft.PowerFx.Json/Functions/CopilotFunctionImpl.cs [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class CopilotFunctionImpl : CopilotFunction, IAsyncTexlFunction5
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Json/Functions/CopilotFunctionImpl.cs:24]: Re using IAsyncTexlFunction5 ....  can we build on #2812  to avoid this?

Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Json/Functions/CopilotFunctionImpl.cs:24]: That's in interpreter... Fx.Json doesn't depend on it. I thought about it... Ideal thing would be having common abstraction library/moving it to core as a compromise...

File: src/tools/Repl/COPILOT_README.md (hunk 2 above)
Comment (jas-valgotar) [reply src/tools/Repl/COPILOT_README.md]: updated

File: src/libraries/Microsoft.PowerFx.Json/Functions/CopilotFunctionImpl.cs (hunk 3 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Json/Functions/CopilotFunctionImpl.cs:24]: Ah ... IAsyncTexlFunction5 didn't get converted over; and that's used for Fx.Json. And this lives in Fx.Json because the implementation needs a json parser to consuem the Gpt result.

**Review by MikeStall — APPROVED** [review]

**Comment by jas-valgotar** [comment]
✅ No public API change.
